pytomo3d/window/window.py

```diff
@@ -58,8 +59,81 @@ def load_window_config_yaml(filename):
     return pyflex.Config(**data)
 
 
+def update_user_levels(user_module, config, station, event, obsd, synt):
+    """Update user levels as an array using user_module
+
+    This function emulates the user_functions feature in FLEXWIN. User
+    has to prepare a python module that has a method called
+    generate_user_levels. This module has to be passed out as a string
+    (It should be formatted as you would in an import statement). And
+    then, user_module is dynamically imported and the
+    generate_user_levels function is called with the following
+    parameters: pyflex config, obspy station data, obspy event data,
+    observed and synthetic obspy traces. It is excepted that user to
+    use given scalar acceptance levels as base values and generate
+    acceptance level arrays as long as trace data to be given to the
+    pyflex. Function should return these values in order: stalta water
+    level, tshift acceptance level, dlna acceptance level,
+    cc_acceptance_level, s2n_limit.
+
+    :param user_module: user module as a string
+    :type user_module: str
+    :param config: window selection config
+    :type config_dict: pyflex.Config
+    :param station: station information which provids station location to
+        calculate the epicenter distance
+    :type station: obspy.Inventory or pyflex.Station
+    :param event: event information, providing the event information
+    :type event: pyflex.Event, obspy.Catalog or obspy.Event
+    :param observed: observed trace
+    :type observed: obspy.Trace
+    :param synthetic: synthetic trace
+    :type synthetic: obspy.Trace
+    :return: window selection config with arrays
+    :rtype: pyflex.Config
+    """
+    # Ridvan Orsvuran, 2016
+    # If user gives a user_module, allow it to create the user
+    # acceptance levels as arrays.
+    try:
+        # Add current working directory to path. This enables user to
+        # use python files in the working directory.
+        import sys
+        sys.path.append(".")
+
+        # Import the user module
+        user = importlib.import_module(user_module)
+        # Assign generate function to a variable. This enables to
+        # catch the AttributeError.
+        generate = user.generate_user_levels
+    except ImportError:
+        raise Exception("Could not import the user_function module:",
+                        user_module)
+    except AttributeError:
+        raise Exception("Given user module does not have a"
+                        "generate_user_levels method.")
+
+    # do not give the original config to the user
+    config_copy = copy.deepcopy(config)
+    stalta_waterlevel, tshift, dlna, cc, s2n = generate(config_copy,
```

Comment:
a typo here? so `generate` should be `generate_user_levels`?

I think I understand what you mean. I will make slightly modifications based on your commit.